### PR TITLE
#153 | Widget automated tests fixed and refactored

### PR DESCRIPTION
**Pull request by szymon-owczarzak**
Trying to fix this annoying bug that causes our tests to fail randomly.

---
I hereby agree to the terms of the Cogboard Contributor License Agreement.

### CODE REVIEW

File: functional/cypress-tests/cypress/support/widgetDynamicTab.js [hunk 1]
```diff
@@ -185,50 +185,53 @@ export function fillWorldClock() {
     .click();
 }
 
-export function fillDynamicTab(type = 'Text', version = '') {
-  const name = `${type}${version}`;
-  if (name !== 'White Space' && name !== 'Checkbox') {
```
Comment (szymon-owczarzak) [inline functional/cypress-tests/cypress/support/widgetDynamicTab.js]: Lets get rid of this if so we don't need to change this in future

File: functional/cypress-tests/cypress/support/widget.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import { fillDynamicTab } from './widgetDynamicTab'
+
+class Widget {
```
Comment (szymon-owczarzak) [inline functional/cypress-tests/cypress/support/widget.js]: Encapsulates widget configuration, and widget related assertions.

File: functional/cypress-tests/cypress/integration/widgets.js [hunk 3]
```diff
@@ -14,40 +13,24 @@ describe('Widgets', () => {
     cy.clickAddWidgetButton();
   });
 
-  for (let i = 0; i < widgetsKeys.length; i++) {
-    const widget = widgetsKeys[i];
-    const name = Widgets[widget].name;
-    const title = `Test-${name}`;
-    let version = '';
-    if (name == 'SonarQube') {
```
Comment (szymon-owczarzak) [inline functional/cypress-tests/cypress/integration/widgets.js]: Lets get rid of this if and make tests more generic

File: functional/cypress-tests/cypress/fixtures/Widgets.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ module.exports = {
   aemHealthcheck: {
     name: 'AEM Healthcheck',
     endpoint: 'endpoint1',
-    schedulePeriod: '500',
+    schedulePeriod: '3',
```
Comment (marcinkp) [inline functional/cypress-tests/cypress/fixtures/Widgets.js]: does it mean widget is refreshed each 3 millisecond during the test?

File: functional/cypress-tests/cypress/support/widget.js (hunk 2 above)
Comment (marcinkp) [reply functional/cypress-tests/cypress/support/widget.js]: maybe it is worth mention about it in readme.

File: functional/cypress-tests/cypress/fixtures/Widgets.js (hunk 4 above)
Comment (szymon-owczarzak) [reply functional/cypress-tests/cypress/fixtures/Widgets.js]: every 3 seconds until it is deleted

File: functional/cypress-tests/cypress/support/widget.js (hunk 2 above)
Comment (szymon-owczarzak) [reply functional/cypress-tests/cypress/support/widget.js]: Yes that is true and i will, but after i finish refactoring - this is just a part of refactor not the final structure.
I want to unblock all hanging PR for now.

File: functional/cypress-tests/cypress/fixtures/Widgets.js (hunk 4 above)
Comment (michaldziubek) [reply functional/cypress-tests/cypress/fixtures/Widgets.js]: Are you sure it is 3 seconds? Does it mean it was set to 500 seconds previously?

Comment (szymon-owczarzak) [reply functional/cypress-tests/cypress/fixtures/Widgets.js]: Yes I'm sure.

**Pull request merged**